src/dbnode/storage/bootstrap/bootstrapper/persist.go

```diff
@@ -259,7 +263,7 @@ func BuildBootstrapIndexSegment(
 		}))
 	}
 
-	indexBlock, ok := indexResults[xtime.ToUnixNano(blockStart)]
+	indexBlockByVolumeType, ok := indexResults[xtime.ToUnixNano(blockStart)]
```

Comment:
Do you want to move `if len(builder.Docs()) == 0 {` to the top of this function too?